exercises/simple-linked-list/src/example/java/SimpleLinkedList.java

```diff
@@ -0,0 +1,86 @@
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+public class SimpleLinkedList<T> {
+
+    public class Element<T> {
+
+        T value;
+        Element next;
+
+        Element(T value) {
+            this.value = value;
+        }
+    }
+
+    private Element<T> head;
+    private int size;
+
+    public SimpleLinkedList() {
+    }
+
+    public SimpleLinkedList(T[] values) {
+        for (int ii = values.length - 1; ii >= 0; ii--) {
+            push(values[ii]);
+        }
+    }
+
+    public void push(T value) {
+        Element<T> newElement = new Element<>(value);
+        this.size++;
+        if (Objects.isNull(head)) {
+            head = newElement;
+        } else {
+            newElement.next = head;
+            head = newElement;
+        }
+    }
+
+    public T pop() {
+        if (Objects.isNull(head)) {
+            throw new NoSuchElementException();
+        }
+        T value = head.value;
+        head = head.next;
+        this.size--;
+        return value;
+    }
+
+    public void reverse() {
+        Element current = head;
+        Element next;
+        Element previous = null;
+        while (Objects.nonNull(current)) {
+            next = current.next;
+            current.next = previous;
+            previous = current;
+            current = next;
+        }
+        head = previous;
+    }
+
+    public T[] asArray(Class<T> clazz) {
+        T[] result = getArray(clazz, this.size);
+        int index = 0;
+        Element<T> current = head;
+        while (Objects.nonNull(current)) {
+            result[index++] = current.value;
+            current = current.next;
+        }
+        return (T[]) result;
```

Comment:
Yes, it seemed to be earlier, but it must have been something I hadn't set up correctly with the types.